mcs/class/System/Mono.Net.Security/MonoTlsStream.cs

```diff
@@ -154,5 +156,21 @@ internal async Task<Stream> CreateStream (WebConnectionTunnel tunnel, Cancellati
 			throw new PlatformNotSupportedException (EXCEPTION_MESSAGE);
 #endif
 		}
+
+		void Dispose (bool disposing)
+		{
+			if (Interlocked.CompareExchange (ref disposed, 1, 0) != 0)
```

Comment:
do we expect this to be called by multiple threads or why are you doing this vs. a simple `if (disposed) return;` check?

Reply:
I'm using this as a pattern in the other related files - does it hurt to have the interlocked here?